staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -272,37 +340,53 @@ func (cfgCtlr *configController) processNextWorkItem() bool {
 
 	func(obj interface{}) {
 		defer cfgCtlr.configQueue.Done(obj)
-		if !cfgCtlr.syncOne() {
+		report := cfgCtlr.SyncOne(map[string]string{})
+		switch {
+		case report.NeedRetry:
 			cfgCtlr.configQueue.AddRateLimited(obj)
-		} else {
+		case report.NeededSpecificWait > 0:
+			cfgCtlr.configQueue.AddAfter(obj, report.NeededSpecificWait)
+		default:
 			cfgCtlr.configQueue.Forget(obj)
 		}
 	}(obj)
 
 	return true
 }
 
-// syncOne attempts to sync all the API Priority and Fairness config
-// objects.  It either succeeds and returns `true` or logs an error
-// and returns `false`.
-func (cfgCtlr *configController) syncOne() bool {
+// SyncReport describes the outcome of a call to SyncOne
+type SyncReport struct {
```

Comment:
It was a bit hard for me to understand why exactly we need this this (including if/what kind of dependency is between those its two fields).

I would suggest to switching SyncOne method to simply return:

```
func SyncOne(...) (backoffDelay time.Duration, err error) {...}
```

This is more aligned with golang style of returning error and I think it makes handling code more intuitive, i.e.:
```
backoffDelay, err := syncOne(...)
if err != nil {
  cfgCtlr.configQueue.AddRateLimited(obj)
  return
}
if backoffDelay > 0 {
  cfgCtlr.configQueue.AddAfter(backoffDelay, obj)
} else 
  cfgCtlr.configQueue.Forget(obj)
}
```

Reply:
done